Release v1.5.6

yakov-g opened the pull request:
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
- add error message when trying to run perf-test for unstructured mode
- RAPTOR-5328: set HOME to /opt and WORKDIR to /opt/code for every env
- add a memo regarding DRUM usage in Windows
- split julia env image into base image and env itself

## Rationale

zohar-mizrahi commented on custom_model_runner/README.md:
```diff
@@ -8,6 +8,8 @@ DRUM can also:
 - perform model validation tests, e.g., checking model functionality on corner cases, like null values imputation.
 - run models in a Docker container.
 
+***DataRobot DRUM is only tested to support Linux/macOS operating systems. To run DRUM in WIndows 10 please use WSL (Windows Subsystem for Linux).***
```
nit: `Windows 10` (capital `I`)

zohar-mizrahi commented on custom_model_runner/datarobot_drum/drum/args_parser.py:
```diff
@@ -920,6 +920,12 @@ def verify_options(options):
+                options.subparser_name == ArgumentsOptions.PERF_TEST
+                and options.target_type == TargetType.UNSTRUCTURED.value
+            ):
+                print("Performance testing is not implemented for the unstructured models.")
```
nit: I'd remove the word `the`, as it's a kind of general message

zohar-mizrahi commented on public_dropin_environments/julia_mlj/Dockerfile:
```diff
@@ -36,10 +35,11 @@ RUN julia "$HOME/julia/sysim.jl" && \
-ENV CODE_DIR=/opt/code 
-ENV ADDRESS=0.0.0.0:8080 WITH_ERROR_SERVER=1
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
```suggestion
WORKDIR ${CODE_DIR}
```

zohar-mizrahi commented on public_dropin_environments/java_codegen/Dockerfile:
```diff
@@ -14,7 +14,9 @@ RUN pip3 install -U --upgrade-strategy eager --no-cache-dir -r dr_requirements.t
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
```suggestion
WORKDIR ${CODE_DIR}
```

zohar-mizrahi commented on public_dropin_environments/julia_mlj/Dockerfile:
```diff
@@ -36,10 +35,11 @@ RUN julia "$HOME/julia/sysim.jl" && \
-ENV PRODUCTION=1 MAX_WORKERS=1 SHOW_STACKTRACE=1
+#ENV PRODUCTION=1 MAX_WORKERS=1 SHOW_STACKTRACE=1
 
 ENTRYPOINT ["/opt/code/start_server.sh"]  
```
```suggestion
ENTRYPOINT ["${CODE_DIR}/start_server.sh"]  
```

zohar-mizrahi commented on public_dropin_environments/python3_keras/Dockerfile:
```diff
@@ -20,7 +20,8 @@ RUN pip3 install -r requirements.txt --no-cache-dir && \
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
```suggestion
WORKDIR ${CODE_DIR}
```

zohar-mizrahi commented on public_dropin_environments/python3_pmml/Dockerfile:
```diff
@@ -22,7 +22,8 @@ RUN pip3 install -r requirements.txt --no-cache-dir && \
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
```suggestion
WORKDIR ${CODE_DIR}
```

zohar-mizrahi commented on public_dropin_environments/python3_pytorch/Dockerfile:
```diff
@@ -20,7 +20,8 @@ RUN pip3 install -r requirements.txt --no-cache-dir && \
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
```suggestion
WORKDIR ${CODE_DIR}
```

zohar-mizrahi commented on public_dropin_environments/python3_xgboost/Dockerfile:
```diff
@@ -20,7 +20,8 @@ RUN pip3 install -r requirements.txt --no-cache-dir && \
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
```suggestion
WORKDIR ${CODE_DIR}
```

zohar-mizrahi commented on public_dropin_environments/r_lang/Dockerfile:
```diff
@@ -13,7 +13,8 @@ RUN pip3 install -U --upgrade-strategy eager -r dr_requirements.txt --no-cache-d
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
```suggestion
WORKDIR ${CODE_DIR}
```

yakov-g commented:
jarvis test cmrunner_integration

meltzg reviewed (changes requested):
what are all these new files related to julia? were they meant to be added in this PR?

yakov-g commented:
jarvis test functional_inference_model_templates

yakov-g commented:
> what are all these new files related to julia? were they meant to be added in this PR?

We already have julia env in the repo, but it takes more than timeout to build this env during the test. So I had to split it into a base image and public env image itself.
Just removed files i moved to the base image from the public env.

timsetsfire reviewed (approved):
looks great!
